src/components/ecl-button/ecl-button.tsx

```diff
@@ -19,40 +19,74 @@ export class EclButton {
   @Prop() hideLabel: boolean = false;    
   @Prop() ariaControls: string;
   @Prop() itemId: string;
+  @Prop() showIndicator: boolean = false;
+  @Prop() indicatorValue: string = '';
+  @State() indicatorSlot: 'before' | 'after' = 'after';
+  @State() hasLabelContent: boolean = false;
+  @State() hasIconBefore: boolean = false;
+  @State() hasIconAfter: boolean = false;
 
-  componentDidRender() {
+  componentDidLoad() {
     const dataAttrs = Object.keys(this.el.dataset);
     if (dataAttrs) {
       dataAttrs.forEach((attr) => {
-        const button = this.el.firstElementChild as HTMLElement;
+        const button = this.el.querySelector('button');
         const attrValue = this.el.dataset[attr];
         if (attrValue === '' || attrValue === attr) {
-          button.dataset[attr] = attr; // Booleans
+          button.dataset[attr] = attr;
         } else {
           button.dataset[attr] = attrValue;
         }
         delete this.el.dataset[attr];
       });
     }
 
-    if (this.el.getElementsByTagName('ecl-icon')[0] && this.el.querySelector('.ecl-icon')) {
-      const slot = this.el.getElementsByTagName('ecl-icon')[0].getAttribute('slot');
-      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon');
-      if (slot && !this.hideLabel) {
-        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`, `sc-ecl-button-${this.theme}`);
+    const iconEl = this.el.querySelector('ecl-icon');
```

Comment:
tricky, we are still using the icon from v3 here, in v5 we have a span that gets replaced by an external script onLoad with the svg markup, not sure what would happen here, but anyway, at the moment we cannot use the v5 approach, it doesn't work, so ok to keep this logic, just that you changed it to a componentDidLoad which might be even more problematic, specific reason for this? Just asking..

Reply:
rollback how it was, i though an upgrade would be nice here

Reply:
it was just a question.. :) And the right answer would have been that using componentDidRender for then using its internal logic in the render method of the component was probably not a good idea, i see that doing it makes the icon not appear most of the times when you select it for the first time, componentDidLoad works better, for the issue with the script that we are supposed to use we will see, bring back your componentDidLoad ;)